### PR TITLE
feat: TAP-2415 License management needs to be able to control accordi…

### PR DESCRIPTION
**Pull request by mnianqi**
…ng to data source

### CODE REVIEW

File: manager/tm/src/main/java/com/tapdata/tm/ds/service/impl/DataSourceServiceImpl.java
Comment (HarsenLin) [inline manager/tm/src/main/java/com/tapdata/tm/ds/service/impl/DataSourceServiceImpl.java:1]: Whether to change line 1941 to 'options.getDatasourceInstanceInfo()'

Comment (HarsenLin) [inline manager/tm/src/main/java/com/tapdata/tm/ds/service/impl/DataSourceServiceImpl.java:1]: Whether to change line 1941 to 'options.getDatasourceInstanceInfo()'

**Review by HarsenLin — CHANGES_REQUESTED** [review]
Whether to change line 1941 to 'options.getDatasourceInstanceInfo()'

**Comment by xbsura** [comment]
SonarQube Quality Gate Status: **ERROR**

- Status: **ERROR**, MetricKey: **new_coverage**, ActualValue: **60.0**


See Sonar Scan Result at: http://58.251.34.123:29000/dashboard?branch=TAP-2415&id=tapdata

**Review by ply0011 — APPROVED** [review]

**Review by HarsenLin — APPROVED** [review]

**Pull request merged**